django_prometheus/db/backends/postgis/base.py

```diff
@@ -1,16 +1,39 @@
-import psycopg2.extensions
-from django.contrib.gis.db.backends.postgis import base
+from django import VERSION
 
-from django_prometheus.db.common import DatabaseWrapperMixin, ExportingCursorWrapper
+if VERSION < (4, 2):
+    import psycopg2.extensions
+    from django.contrib.gis.db.backends.postgis import base
 
+    from django_prometheus.db.common import DatabaseWrapperMixin, ExportingCursorWrapper
 
-class DatabaseWrapper(DatabaseWrapperMixin, base.DatabaseWrapper):
-    def get_connection_params(self):
-        conn_params = super().get_connection_params()
-        conn_params["cursor_factory"] = ExportingCursorWrapper(psycopg2.extensions.cursor, "postgis", self.vendor)
-        return conn_params
+    class DatabaseWrapper(DatabaseWrapperMixin, base.DatabaseWrapper):
+        def get_connection_params(self):
+            conn_params = super().get_connection_params()
+            conn_params["cursor_factory"] = ExportingCursorWrapper(
+                psycopg2.extensions.cursor, "postgis", self.vendor
+            )
+            return conn_params
 
-    def create_cursor(self, name=None):
-        # cursor_factory is a kwarg to connect() so restore create_cursor()'s
-        # default behavior
-        return base.DatabaseWrapper.create_cursor(self, name=name)
+        def create_cursor(self, name=None):
+            # cursor_factory is a kwarg to connect() so restore create_cursor()'s
+            # default behavior
+            return base.DatabaseWrapper.create_cursor(self, name=name)
+
+else:
+    from django.contrib.gis.db.backends.postgis import base
+    from django.db.backends.postgresql.base import Cursor
+
+    from django_prometheus.db.common import DatabaseWrapperMixin, ExportingCursorWrapper
```

Comment:
the common imports (`from django.contrib.gis.db.backends.postgis import base` and `from django_prometheus.db.common import DatabaseWrapperMixin, ExportingCursorWrapper`) should be at the top of the file.
unless I am missing something, there is no reason to duplicate this import code.

you can also avoid  duplicate  code here by setting a cursor class  using the if statement and using that.
it will look something like this:

```python 
if VERSION < (4, 2):
    from  psycopg2.extension import cursor as cursor_cls
else:
   from django.db.backends.postgresql.base import Cursor as cursor_cls

....
conn_params["cursor_factory"] = ExportingCursorWrapper(cursor_cls, "postgis", self.vendor) 
```

Reply:
very good point, fixed now